src/components/FeatureViews/AdminPortal/UnpaidSessions.js

```diff
@@ -0,0 +1,141 @@
+            ...bindActionCreators(adminActions, dispatch),
+        }),
+        [dispatch]
+    );
+
+    const UnpaidList = useSelector(state => state.Admin.students);
+
+    useEffect(()=>{
+        api.fetchUnpaid();
+    },[]);
+
+    const styles = (username) => ({
+        "backgroundColor": stringToColor(username),
+        "color": "white",
+        "margin": 9,
+        "width": 47,
+        "height": 47,
+        "fontSize": 20,
+        "alignSelf": "center",
+        "fontFamily": "Roboto"
+    });
+
+    const roleStyle = (role) => ({
+        "backgroundColor": roleColor(role),
+        "borderRadius": "50px",
+        "color": "white",
+        "height": 30,
+        "margin": 10,
+        "padding": 5,   
+        "fontSize": 14,
+        "alignSelf": "center",
+        "textAlign": "center"
+    })
+
+    const cardStyle = ()=> ({
+        "height": 250,  
+        "width": 220,
+        "margin": '10px'
+    })
+
+    const statusStyle = (status) => ({
+        "backgroundColor": statusColor(status),
+        "color": "black",
+        "height": 10,
+        "width": 10,
+        "lineHeight": 1,
+        // "lineHeight": "auto", 
+        "margin": "auto",
+        "padding": 10,
+        "borderRadius": "50%",
+        // "alignContent": "center"
+
+    })
+
+const checkUnpaid = (x) => {
+    if (!x){
+        return <Loading/>
+    }
+
+    else {
+        console.log('x exists')
+
+        return x.map(up=>(
+            displayUnpaid(up)
+        ))
+        
+    }
+}
+
+const displayUnpaid = (unpaid) => {
+    console.log(unpaid)
+    return(
+    <Card style = {cardStyle()}> 
+                <CardActionArea>
+                    <CardMedia>
+                        <Grid container style={{justifyContent:"center"}}>
+                            <Avatar   
+                                    style={styles(unpaid.fName + " " + unpaid.lName)}
+                                    >
+                                        {initials(unpaid.fName, unpaid.lName)}
+                            </Avatar>
+                        </Grid>
+                    </CardMedia>
+                    <CardContent>
+                    <Typography style ={{fontSize: "16px", fontWeight: 500, lineHeight: "24px", textAlign: "center"}}>
+                        {unpaid.fName + " " + unpaid.lName}
+                    </Typography>
+                    <Typography style={roleStyle(unpaid.status)} >
+                        {parseRole(unpaid.status)}
+                    </Typography>
+                    <Typography style={{textAlign: "center"}}>
+                        Payment Status: <span style={statusStyle(unpaid.paymentStatus)}>{unpaid.paymentStatus}</span>
+                        <br></br>
+                        {unpaid.amt}
+                        <br></br>
+                        {unpaid.course}
+                    </Typography>
+                    </CardContent>
+                </CardActionArea>
+            </Card>
+    )
+}
```

Comment:
Move this to its own file and component, to avoid this file becoming too large later on.